zebra-state/src/service/finalized_state/disk_format/upgrade/fix_tree_key_type.rs

```diff
@@ -0,0 +1,138 @@
+//! Updating the sprout and history tree key type from `Height` to the empty key `()`.
+//!
+//! This avoids a potential concurrency bug, and a known database performance issue.
+
+use std::sync::{mpsc, Arc};
+
+use zebra_chain::{block::Height, history_tree::HistoryTree, sprout};
+
+use crate::service::finalized_state::{
+    disk_db::DiskWriteBatch, disk_format::MAX_ON_DISK_HEIGHT, ZebraDb,
+};
+
+use super::CancelFormatChange;
+
+/// Runs disk format upgrade for changing the sprout and history tree key types.
+///
+/// Returns `Ok` if the upgrade completed, and `Err` if it was cancelled.
+#[allow(clippy::unwrap_in_result)]
+#[instrument(skip(upgrade_db, cancel_receiver))]
+pub fn run(
+    initial_tip_height: Height,
```

Comment:
###### Optional Suggestion

```suggestion
    _initial_tip_height: Height,
```